src/Block.cpp

```diff
@@ -491,6 +492,19 @@ Block::append_nested_loop(CGContext& cg_context)
 	return sf;
 }
 
+/* Make sure the vector and the set contain the same elements, under the
+ * assumption that the vector can only ever grow. If the sizes of the
+ * containers are equal, this function assumes that they do contain the same
+ * elements. Otherwise, the elements of the vector are added to the set.
+ */
+static void
+ensure_vector_and_set_match(const vector<Variable *>& vec, set<Variable *>& set)
+{
+	if (vec.size() != set.size()) {
```

Comment:
This becomes unnecessary if we don't maintain both a vector and a set for local variables.